packages/@aws-cdk/aws-appsync/lib/graphqlapi.ts

```diff
@@ -429,15 +410,15 @@ export class GraphQLApi extends GraphqlApiBase {
     this.arn = this.api.attrArn;
     this.graphQlUrl = this.api.attrGraphQlUrl;
     this.name = this.api.name;
-    this.schemaMode = props.schemaDefinition;
-    this.schema = this.defineSchema(props.schemaDefinitionFile);
+    this.schema = props.schema ?? new Schema();
+    this._schema = this.schema.bind(this);
 
     if (modes.some((mode) => mode.authorizationType === AuthorizationType.API_KEY)) {
       const config = modes.find((mode: AuthorizationMode) => {
         return mode.authorizationType === AuthorizationType.API_KEY && mode.apiKeyConfig;
       })?.apiKeyConfig;
       this._apiKey = this.createAPIKey(config);
-      this._apiKey.addDependsOn(this.schema);
+      this._apiKey.addDependsOn(this._schema);
```

Comment:
How about a more informative description fore the lower level construct?
`this.apiKeyResource` instead of `this._apiKey`